resolver-dns/src/main/java/io/netty/resolver/dns/DnsQueryContext.java

```diff
@@ -53,12 +57,15 @@
 
     DnsQueryContext(DnsNameResolver parent,
                     InetSocketAddress nameServerAddr,
-                    DnsQuestion question, Promise<AddressedEnvelope<DnsResponse, InetSocketAddress>> promise) {
-
-        this.parent = parent;
-        this.nameServerAddr = nameServerAddr;
-        this.question = question;
-        this.promise = promise;
+                    DnsQuestion question,
+                    List<DnsRecord> additional,
+                    Promise<AddressedEnvelope<DnsResponse, InetSocketAddress>> promise) {
+
+        this.parent = checkNotNull(parent, "parent");
+        this.nameServerAddr = checkNotNull(nameServerAddr, "nameServerAddr");
+        this.question = checkNotNull(question, "question");
+        this.additional = checkNotNull(additional, "additional");
```

Comment:
Could we convert `additional` to an array so that we are safe from an unexpected modification of `additional` by a user?